test/src/androidTest/kotlin/io/realm/shared/notifications/RealmListNotificationsTests.kt

```diff
@@ -33,38 +47,208 @@ class RealmListNotificationsTests : NotificationTests {
     }
 
     @Test
-    @Ignore
     override fun initialElement() {
-        TODO("Waiting for RealmList support")
+        val dataSet = OBJECT_VALUES
+
+        realm.writeBlocking {
+            val managedContainer = copyToRealm(RealmListContainer())
+            val managedDataSet = dataSet.map { copyToRealm(it) }
+            managedContainer.objectListField
+                .addAll(managedDataSet)
+        }
+
+        runBlocking {
+            val channel = Channel<RealmList<*>>(capacity = 1)
+            val observer = async {
+                val container = realm.objects<RealmListContainer>()
+                    .first()
+
+                container.objectListField
+                    .observe()
+                    .collect { flowList ->
+                        channel.send(flowList)
+                    }
+            }
+
+            // Assertion after empty list is emitted
+            val emittedList = channel.receive()
+            assertNotNull(emittedList)
+            assertEquals(dataSet.size, emittedList.size)
+
+            observer.cancel()
+            channel.close()
+        }
     }
 
     @Test
-    @Ignore
     override fun observe() {
-        TODO("Waiting for RealmList support")
+        val dataSet = OBJECT_VALUES
+
+        realm.writeBlocking {
+            // Just create an empty container with empty lists
+            copyToRealm(RealmListContainer())
+        }
+
+        runBlocking {
+            val channel = Channel<RealmList<*>>(capacity = 1)
+            val observer = async {
+                val container = realm.objects<RealmListContainer>()
+                    .first()
+
+                container.objectListField
+                    .observe()
+                    .collect { flowList ->
+                        channel.send(flowList)
+
+                        realm.writeBlocking {
```

Comment:
Why not do the write imperatively in between the asserts instead of here?

Reply:
That's right, I'm still getting used to using channels as blocking mechanisms 🐒